oio/common/service_resolver.py

```diff
@@ -0,0 +1,72 @@
+# Copyright (C) 2015-2019 OpenIO SAS, as part of OpenIO SDS
+#
+# This library is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 3.0 of the License, or (at your option) any later version.
+#
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this library.
+
+
+from urlparse import urlparse, urlunparse
+
+from oio.common.green import time
+from oio.common.logger import get_logger
+from oio.conscience.client import ConscienceClient
+
+
+class ServiceResolver(object):
+    """A caching client to rawx services."""
+
+    def __init__(self, conf, max_age=60, logger=None):
+        self.conf = conf
+        self.logger = logger or get_logger(conf)
+        self.conscience_client = ConscienceClient(self.conf, logger=logger)
```

Comment:
I have a concern with this class having its own instance of `ConscienceClient` (with its own set of connections to the proxy). Actually I think `ServiceResolver` could be merged with `ConscienceClient`.